File: app/src/main/java/com/mdgd/pokemon/ui/pokemon/items/TitlePropertyData.java
package com.mdgd.pokemon.ui.pokemon.items;

public class TitlePropertyData implements TitleProperty {

    private final int titleResId;
    private final int nestingLevel;

    public TitlePropertyData(int titleResId) {
        this(titleResId, 0);
    }

    public TitlePropertyData(int titleResId, int nestingLevel) {
        this.titleResId = titleResId;
        this.nestingLevel = nestingLevel;
    }

    @Override
    public int getTitleResId() {
        return titleResId;
    }

    @Override
    public int getType() {
        return PROPERTY_TITLE;
    }

    @Override
    public int getNestingLevel() {
        return nestingLevel;
    }
}

File: app/src/main/java/com/mdgd/pokemon/ui/pokemon/dto/TitlePropertyData.kt
package com.mdgd.pokemon.ui.pokemon.dto

open class TitlePropertyData @JvmOverloads constructor(override val titleResId: Int, override val nestingLevel: Int = 0) : TitleProperty {
    override val type: Int
        get() = PokemonProperty.PROPERTY_TITLE
}
